source/common/grpc/common.cc

```diff
@@ -24,13 +24,63 @@ namespace Grpc {
 
 const std::string Common::GRPC_CONTENT_TYPE{"application/grpc"};
 
-void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& grpc_service,
-                        const std::string& grpc_method, bool success) {
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& protocol,
+                        const std::string& grpc_service, const std::string& grpc_method,
+                        const Http::HeaderEntry* grpc_status) {
+  if (!grpc_status) {
+    return;
+  }
+  uint64_t grpc_status_code;
```

Comment:
You can use https://github.com/grpc/grpc/blob/5253c8f9a899450397a5e46e4923d01ac9a66a27/include/grpc%2B%2B/impl/codegen/status_code_enum.h
As envoy is already depends on gRPC lib.
However, if the plan is to alway use envoy HTTP2, and remove the gRPC lib deps from envoy, we need to fork that enum and keep them in sync.
Here, I don't need an enum, as the stat tracking just need an int.